group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/OptimizedUniformAssignmentBuilder.java

```diff
@@ -71,63 +69,54 @@ public class OptimizedUniformAssignmentBuilder extends AbstractUniformAssignment
      */
     private final Set<Uuid> subscribedTopicIds;
 
-    /**
-     * The number of members to receive an extra partition beyond the minimum quota.
-     * Minimum Quota = Total Partitions / Total Members
-     * Example: If there are 11 partitions to be distributed among 3 members,
-     *          each member gets 3 (11 / 3) [minQuota] partitions and 2 (11 % 3) members get an extra partition.
-     */
-    private int remainingMembersToGetAnExtraPartition;
-
     /**
      * Members mapped to the remaining number of partitions needed to meet the minimum quota.
-     * Minimum quota = total partitions / total members.
      */
-    private Map<String, Integer> potentiallyUnfilledMembers;
+    private final List<MemberWithRemainingQuota> potentiallyUnfilledMembers;
```

Comment:
why do we call this potentiallyUnfilledMembers rather than unfilledMembers?

Reply:
There is a set number of extra partitions that we can give out once every member gets an equal number of partitions. For exmaple if there were 5 partitions and 3 members. Each member gets 1 each and then we have 2 extras. Let's say A,B,C have all received 1 partition, they are all potentially unfilled. Once A and B get their extra partitions, C isn't unfilled and doesn't receive any partitions